Sources/Player/PlaybackEngine/Internal/Events/AsyncScheduler.swift

```diff
@@ -0,0 +1,23 @@
+protocol AsyncScheduler {
+  static func schedule(code: @escaping () async -> Void) -> AsyncScheduler
+}
+
+protocol AsyncSchedulerFactory {
+  @discardableResult
+  func create(code: @escaping () async -> Void) -> AsyncScheduler
+}
+
+extension Task: AsyncScheduler {
+  static func schedule(code: @escaping () async -> Void) -> AsyncScheduler {
+	Task<Void, Never> {
```

Comment:
[suggestion] I think we can use our typealias `SafeTask`. Same below.
[suggestion 2] I think formatting is missing, I think you just need to run the setup again.
```suggestion
	SafeTask {
```